### PR TITLE
refactor: mfa client updates

### PR DESCRIPTION
**Pull request by kangmingtay**
## What kind of change does this PR introduce?
* merged `rc` branch in and resolved the merge conflicts 
* use the `decodeBase64URL` helper function in the `rc` branch

### CODE REVIEW

File: src/GoTrueClient.ts [hunk 1]
```diff
@@ -1114,7 +1153,7 @@ export default class GoTrueClient {
           throw error
         }
         // Default to Authorization header if there is no existing session
-        jwt = data.session?.access_token ?? this.headers['Authorization']
+        jwt = data.session?.access_token ?? this.headers['Authorization'].split(' ')[1]
```
Comment (kangmingtay) [inline src/GoTrueClient.ts:1156]: need to split the header because the header also contains `Bearer` in it

File: src/GoTrueClient.ts [hunk 2]
```diff
@@ -1141,7 +1180,7 @@ export default class GoTrueClient {
...
```
Comment (kangmingtay) [inline src/GoTrueClient.ts:1183]: ditto here

File: test/GoTrueClient.test.ts [hunk 3]
```diff
@@ -221,19 +238,21 @@ describe('GoTrueClient', () => {
...
-    })
+      const { error, data } = await auth.signUp({
+        email,
+        password,
+      })
 
-    expect(error).toBeNull()
-    expect(data.session).not.toBeNull()
-    expect(data.user).not.toBeNull()
+      expect(error).toBeNull()
```
Comment (J0) [inline test/GoTrueClient.test.ts:250]: is the extra space intended?

**Review by J0 — APPROVED** [review]

File: test/GoTrueClient.test.ts (hunk 3 above)
Comment (kangmingtay) [reply test/GoTrueClient.test.ts:250]: extra space here is due to it being wrapped by line 242

**Pull request merged**